### PR TITLE
[Entity Features] Handle Enter

### PR DESCRIPTION
**Pull request by juliaroldi**
Handle some edge cases for enter key behavior on entity and add `ClickOnEntityFeature` for keyboard event

Edge cases fixed: 
1) Before enitity when is the first element of segment 
2) After enitity when is the last element of segment 

TEST_ENTITY

| Feature    | State |
| -------- | ------- |
| `ClickOnEntityFeature` | Added    |
| `EscapeFromEntityFeature` | Removed     |
| `EnterBeforeReadonlyEntityFeature`    |  Feature already handled by Entity Plugin, only edge case was fixed   |
| `DeleteBeforeEntityFeature`    | Feature already handled by Entity Plugin, only edge case was fixed     |
| `MoveBetweenDelimitersFeature`    | Feature already handled by Entity Plugin    |
| `RemoveEntityBetweenDelimitersFeature`    | Feature already handled by Entity Plugin    |

### CODE REVIEW

**Comment by JiuqingSong** [comment]
What content edit features can be turned off after this change?

File: packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts [hunk 1]
```diff
@@ -241,6 +243,14 @@ export function handleDelimiterKeyDownEvent(editor: IEditor, event: KeyDownEvent
                 }
             }
         }
+    } else {
+        const entity = getSelectedEntity(selection);
+        if (entity && isNodeOfType(entity, 'ELEMENT_NODE') && helper.isNodeInEditor(entity)) {
+            selection.range.selectNode(entity);
+            if (isEnter) {
+                triggerEntityEventOnEnter(editor, entity, rawEvent);
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:251]: If this is only for ENTER key, check ENTER key first, before searching the entity. That will have better performance

**Comment by juliaroldi** [comment]
> What content edit features can be turned off after this change?

After that change, we only need to port markdown features

File: packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts (hunk 1 above)
Comment (juliaroldi) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:251]: The entity is being searched to be selected when the range is expanded, then when we pressed any key, the key pressed can be inserted in the range of the entity. This is was being handle by `EnterBeforeReadonlyEntityFeature`, but was not handle by the EntityPlugin.

Comment (JiuqingSong) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:251]: hmm. ok. Is this moved from some old code? If so, where is the old code?

Comment (juliaroldi) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:251]: it is not moved from old code, but here is the code to remove the expanded selection 
Screenshot 2024-03-19 at 14 21 58

Comment (juliaroldi) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:251]: Maybe we should try only delete the selection content and do not prevent default behavior. I'll try this.

File: packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts [hunk 2]
```diff
@@ -303,3 +320,31 @@ export const handleEnterInlineEntity: ContentModelFormatter = model => {
...
+                type: format.entityType,
+                isReadonly: !!format.isReadonly,
+                wrapper,
+            },
+            rawEvent: rawEvent,
+        });
+    }
+};
+
+const getSelectedEntity = (selection: RangeSelection) => {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:344]: In PR https://github.com/microsoft/roosterjs/pull/2492, @florian-msft  is adding a method to get current entity from a given node. I think you can wait and leverage the function

Comment (JiuqingSong) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts:344]: image

Comment (JiuqingSong) [reply packages/roosterjs-content-model-core/lib/corePlugin/entity/entityDelimiterUtils.ts]: @juliaroldi the PR is merged, you can use the new API now.

File: packages/roosterjs-content-model-dom/lib/domUtils/entityUtils.ts [hunk 3]
```diff
@@ -131,6 +132,15 @@ export function addDelimiters(
     return [delimiterAfter, delimiterBefore];
 }
 
+/**
+ * Find the closest entity wrapper element from a given DOM node
+ * @param node The node to start looking for entity wrapper
+ * @param domHelper The DOM helper to use
+ */
+export function findClosestEntityWrapper(node: Node, domHelper: DOMHelper): HTMLElement | null {
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-dom/lib/domUtils/entityUtils.ts:140]: there is already such function, please pull to latest and resolve the conflict

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**